### ISSUE TITLE
Not letting HA to fall asleep

### ISSUE THREAD

**Issue by Borales** [seaky/nspanel_pro_tools_apk#150]
I have a device with HA + [clock](https://www.apkmirror.com/apk/google-inc/clock/) (screensaver) installed. 

NSPanel Pro tool settings:
- Display sleep: 15 sec
- Display sleep mode: Off
- Also "Autostart" is configured to launch HA (w/ watchdog). 

Native android settings:
- "sleep after 15 sec of inactivity"
- "Clock" screensaver setup.

Battery opt is turned off for HA app. 

After a few hours of no activity - HA app falls asleep. So when you wake up the device - AH boots up again (that takes >10 sec and sometimes the device goes into a sleep mode again).

Is there a way to prevent this?

**Comment by Borales** [comment]
I kinda solved this by using MacroDroid app (or I guess any automation app would do the trick too) and adding an automation to turn on the screen every 1.5h (I believe the HA sleeps after ~2h of inactivity).

![screen](https://github.com/user-attachments/assets/305c3f0e-44cc-419d-bc39-072365619671)

so @seaky if this app is not designed/meant for keeping the "autostart app" alive - feel free to close this issue.

**Comment by seaky** [comment]
Indeed, something wierd if I do not use screensaver HA lasts forever if I turn on after some our the HA process dies.

**Comment by seaky** [comment]
btw that is the purpose of the watchdog feature to keep autostart app alive so I have to investigate

**Comment by seaky** [comment]
I've been testing with no screensaver active for 2 days and HA still active so definitely the screensaver cases this issue
